azerothcore/azerothcore-wotlk
# fix(Core/CharacterHandler): Updated racial skills to be deleted on faction/race change

Racial skills are not being deleted on faction/race changes, resulting in the following server error message when players login after race/faction changes:

`Character GUID Full: {} Type: Player Low: {} has skill {} that does not exist.`

If a player changes race multiple times, they will accumulate multiple unusable racial skills in the database resulting in multiple error messages as above.

The changes in this commit update the player skills on race/faction change to prevent this behaviour.

## Changes Proposed:
This PR proposes changes to:

## Tests Performed:
This PR has been:

## How to Test the Changes:

1. Create a character of any race, e.g. night elf.
2. Trigger a race-change, and select a different race.
3. Without this PR, the following warning message will appear in the server console:

```Character GUID Full: {} Type: Player Low: {} has skill {} that does not exist.```

4) With this PR, no error messages are reported because the unusable skills were removed.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
